ee/modules/600-flant-integration/hooks/madison/registration.go

```diff
@@ -32,47 +70,96 @@ const (
 	registrationURL  = connectBaseURL + "/v1/madison_register"
 	connectStatusURL = connectBaseURL + "/v1/madison_status"
 
-	madisonKeyPath = "flantIntegration.madisonAuthKey"
-	licenseKeyPath = "flantIntegration.internal.licenseKey"
+	madisonKeyPath         = "flantIntegration.madisonAuthKey"
+	internalMadisonKeyPath = "flantIntegration.internal.madisonAuthKey"
+	internalLicenseKeyPath = "flantIntegration.internal.licenseKey"
+
+	prometheusSecretNS      = "d8-monitoring"
+	prometheusSecretName    = "prometheus-https-mode"
+	prometheusSecretField   = "https_mode"
+	prometheusSecretBinding = prometheusSecretName
+
+	madisonSecretNS      = "d8-monitoring"
+	madisonSecretName    = "madison-proxy"
+	madisonSecretField   = "auth-key"
+	madisonSecretBinding = madisonSecretName
 )
 
+func filterMadisonSecret(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	secret := &v1.Secret{}
+	err := sdk.FromUnstructured(obj, secret)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert secret to struct: %v", err)
+	}
+
+	return string(secret.Data[madisonSecretField]), nil
+}
+
+func filterPrometheusSecret(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	secret := &v1.Secret{}
+	err := sdk.FromUnstructured(obj, secret)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert secret to struct: %v", err)
+	}
+
+	return string(secret.Data[prometheusSecretField]), nil
+}
+
 func registrationHandler(input *go_hook.HookInput, dc dependency.Container) error {
-	_, ok := input.Values.GetOk(madisonKeyPath)
+	// Remove madisonAuthKey if license is not set.
+	licenseKey, ok := input.Values.GetOk(internalLicenseKeyPath)
+	if !ok {
+		input.Values.Remove(internalMadisonKeyPath)
+		return nil
+	}
+
+	// Support existing clusters: restore auth key from configuration.
+	madisonKey, ok := input.ConfigValues.GetOk(madisonKeyPath)
 	if ok {
+		input.Values.Set(internalMadisonKeyPath, madisonKey.String())
 		return nil
 	}
 
-	licenseKey, ok := input.Values.GetOk(licenseKeyPath)
-	if !ok {
+	// Restore auth key from the Secret. (e.g. after restart).
+	if len(input.Snapshots[madisonSecretBinding]) > 0 {
+		storedAuthKey := input.Snapshots[madisonSecretBinding][0].(string)
+		input.Values.Set(internalMadisonKeyPath, storedAuthKey)
 		return nil
 	}
 
+	// Return if auth key is already in values.
+	_, ok = input.Values.GetOk(internalMadisonKeyPath)
+	if ok {
+		return nil
+	}
+
+	// No auth key set in configuration, no auth key stored in the Secret — register in madison.
 	var (
-		domainTemplate  = input.Values.Get("global.modules.publicDomainTemplate").String()
-		globalHTTPSMode = input.Values.Get("global.modules.https.mode").String()
+		domainTemplate      = input.Values.Get("global.modules.publicDomainTemplate").String()
+		globalHTTPSMode     = input.Values.Get("global.modules.https.mode").String()
```

Comment:
I assume global https mode is not required anymore.

Reply:
HTTPS mode detections is deleted.